### DISCUSSION TITLE
Multiplatform UI

### DISCUSSION THREAD

**Discussion by chrpai** [ironfede/openmcdf#277] [General]
Is there any interest in (me and whoever) creating a new Storage Explorer that is multiplatform?  If so, what desired platforms?  Linux, MacOS, iOS, Android?

**Comment by jeremy-visionaid** [comment]
I for one would find that pretty useful! I'd thought about rewriting the explorer using MAUI, and I probably would have done it if there were some obvious de facto standard TreeView-esque control available... The only open source one I saw was:
https://github.com/enisn/UraniumUI/blob/develop/src/UraniumUI.Material/Controls/TreeView.cs

Aside from a quick investigation, I haven't really used Avalonia, but it at least has one out of the box:
https://docs.avaloniaui.net/docs/reference/controls/treeview-1

So yup, if you wanted to contribute something like that, it would be most welcome!

**Comment by Numpsy** [comment]
>  Linux, MacOS, iOS, Android?

Something that worked cross platform on desktop platforms (and maybe even on WASM) would be nice, though I only use it for develpoment assistance so mobile platforms aren;t so interesting for me there.

I've previously wondered about a version in Avalonia, but have never had the time to look at it (i've used Avalonia for a few other small apps on Windows/Linux/mac though)

I think another question about controls that came up in the past is what hex editor controls there are for other toolkits - there are a few about for Avalonia, but I've never used any of them so I can't say what theire abilities are) - e.g. https://github.com/Washi1337/AvaloniaHex or https://github.com/wieslawsoltes/HexView

**Comment by jeremy-visionaid** [comment]
Yeah, the hex editor is probably a bigger problem really - at least a NuGet is available for Avalonia (https://www.nuget.org/packages/AvaloniaHex). The tree view could simply be avoided by using list/collection views instead, which is arguably better for cross-platform anyway.

**Comment by chrpai** [comment]
I looked at Avalonia years ago but didn't adopt it. I get the impression it's much furthor along now.  I've done some Xamarin, MAUI and Blazor and suspect Avalonia is probably a good fit for this.  If everyone agrees I'll fork the repos and create a POC for review.

**Reply by jeremy-visionaid** [reply]
I use MAUI regularly, but I find the criticisms of it to be quite legitimate - it's a reasonably frustrating experience dealing with all the bugs and regressions. I want it to succeed, but even after a few releases and fixing a few bugs upstream myself, the quality feels pretty poor. Moreover, it would be difficult to recommend MAUI if we wanted to have similar/custom controls like in the existing explorer app, and I don't see a need to limit ourselves to the lowest common denominator of native contols. So, yup, I'd think Avalonia probably makes more sense to me (but I also wouldn't veto a MAUI approach - either would be welcome so long as the features are maintained).

**Reply by jeremy-visionaid** [reply]
So yeah, if you write a POC on a branch somewhere, I'd be happy to review it. Thanks in advance for the contribution!

**Comment by chrpai** [comment]
I've started but won't have much for a bit as I'm traveling tomorrow.  We already have a project called StructuredStorageExplorer.  Any ideas on what to name this project?   I've just called it StructuredStorageExplorerPOC for now.

**Comment by chrpai** [comment]
I created a repos and invited you both. https://github.com/chrpai/openmcdf

**Reply by jeremy-visionaid** [reply]
This kind of dropped of my radar a bit, sorry about that... I created #343 to track this as feature. In the meantime, I've done a whole bunch of fixes to the existing explorer if that helps anybody else. Thanks for taking a look at it!
